src/main/java/coresearch/cvurl/io/request/RequestBuilder.java

```diff
@@ -144,14 +146,36 @@ public T logEnabled(boolean logEnabled) {
         return (T) this;
     }
 
+    @SuppressWarnings("unchecked")
+    public T withProxy(CVurlProxy cVurlProxy) {
```

Comment:
It is important to add unit tests for this method

Reply:
Added unit tests.